File: Recursion/binarySearch.java
class binarySearch {

    static void print(int arr[], int s, int e) {
        for (int i = s; i <= e; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println("");
    }

    static boolean BinSearch(int arr[], int s, int e, int key) {

        System.out.print("");
        print(arr, s, e);
        // base case
        if (s > e) {
            return false;
        }

        int mid = s + (e - s) / 2;

        System.out.println("the value of mid " + arr[mid]);

        if (arr[mid] == key) {

            return true;
        }
        if (arr[mid] < key) {
            return BinSearch(arr, mid + 1, e, key);
        } else {
            return BinSearch(arr, s, mid - 1, key);
        }

    }

    public static void main(String[] args) {

        int arr[] = { 2, 4, 6, 10, 14, 18, 22, 38, 49, 222 };
        int size = 10;
        int key = 222;

        Boolean ans = BinSearch(arr, 0, size - 1, key);

        if (ans) {
            System.out.println(" the number is found");
        } else {
            System.out.println(" the number is not found");
        }
    }
}

File: Binary Search/Binary_search.cpp
#include <iostream>
using namespace std;

int binary_search(int arr[], int n, int key)
{
    int f = 0;
    int l = n - 1;
    int mid = (f + l) / 2;

    while (f <= l)
    {
        if (arr[mid] == key)
        {
            return mid;
        }
        else if (key > arr[mid])
        {
            f = mid + 1;
        }
        else if (key < arr[mid])
        {
            l = mid - 1;
        }
        mid = (f + l) / 2;
    }
    return -1;
}

int main()
{
    int ar[7] = {10, 3, 5, 6, 100, 104, 27};
   
    int n = sizeof(ar) / sizeof(ar[0]);
    int index = binary_search(ar, n, 11);
    cout << "index: " << index;
}